examples/applications/plot_time_series_lagged_features.py

```diff
@@ -0,0 +1,433 @@
+# the lagged features in the training set).
+from sklearn.model_selection import TimeSeriesSplit
+
+ts_cv = TimeSeriesSplit(
+    n_splits=3,  # to keep the notebook fast enough on common laptops
+    gap=48,  # 2 days data gap between train and test
+    max_train_size=10000,  # keep train sets of comparable sizes
+    test_size=3000,  # for 2 or 3 digits of precision in scores
+)
+all_splits = list(ts_cv.split(X, y))
+print(len(all_splits))
+
+# %%
+# Training the model and evaluating its performance based on MAPE.
+train_idx, test_idx = all_splits[0]
+X_train, X_test = X.iloc[train_idx], X.iloc[test_idx]
+y_train, y_test = y.iloc[train_idx], y.iloc[test_idx]
+
+model = HistGradientBoostingRegressor().fit(X_train, y_train)
+y_pred = model.predict(X_test)
+mean_absolute_percentage_error(y_test, y_pred)
+
+# %%
+# The error rate of this model is better than our naive shuffling
+# train-test split. This is quite expected but maybe the first split
+# is easier to predict (more regular) than the others. Let's assess this
+# variability of our error evaluation with proper cross-validation:
+from sklearn.model_selection import cross_val_score
+
+cv_mape_scores = -cross_val_score(
+    model, X, y, cv=ts_cv, scoring="neg_mean_absolute_percentage_error"
+)
+cv_mape_scores
+
+# %%
+# The variability across splits is quite large! In a real life setting
+# it would be advised to use more splits to better assess the variability.
+# Let's report the mean CV scores and their standard deviation from now on.
+print(f"CV MAPE: {cv_mape_scores.mean():.3f} ± {cv_mape_scores.std():.3f}")
+
+# To get a finer evaluation of our models we can compute and report several
+# cross-validation metrics at once using a dedicated helper function:
+from time import perf_counter
+from sklearn.model_selection import cross_validate
+from sklearn.metrics import mean_pinball_loss
+from sklearn.metrics import mean_absolute_error
+from sklearn.metrics import mean_squared_error
+
+
+def evaluate(model, X, y, cv):
+    def score_func(estimator, X, y):
+        y_pred = estimator.predict(X)
+        return {
+            "mean_absolute_percentage_error": mean_absolute_percentage_error(y, y_pred),
+            "root_mean_squared_error": np.sqrt(mean_squared_error(y, y_pred)),
+            "mean_absolute_error": mean_absolute_error(y, y_pred),
+            "mean_pinball_05_loss": mean_pinball_loss(y, y_pred, alpha=0.05),
+            "mean_pinball_50_loss": mean_pinball_loss(y, y_pred, alpha=0.50),
+            "mean_pinball_95_loss": mean_pinball_loss(y, y_pred, alpha=0.95),
+        }
+
+    tic = perf_counter()
+    cv_results = cross_validate(
+        model,
+        X,
+        y,
+        cv=cv,
+        scoring=score_func,
+    )
+    toc = perf_counter()
+    for key, value in cv_results.items():
+        if key.startswith("test_"):
+            print(f"{key[5:]}: {value.mean():.3f} ± {value.std():.3f}")

```

Comment:
Instead of printing the results each time, maybe we can add the results to a dataframe in order to give a summary as a table similar to [the one on this example](https://scikit-learn.org/stable/auto_examples/ensemble/plot_gradient_boosting_quantile.html#analysis-of-the-error-metrics).

Please do let us know if you need help on this regard.

Reply:
In particular, this will allow to put the analysis where we compare the impact of the choice of the training loss for each model on the test set evaluation metric in a single place that would not make the reader scroll up and down across sections to compare the numbers for different models on a given metric.